src/main/java/de/rwth/idsg/steve/config/ApiAuthenticationManager.java

```diff
@@ -0,0 +1,116 @@
+ * SteVe - SteckdosenVerwaltung - https://github.com/steve-community/steve
+ * Copyright (C) 2013-2024 SteVe Community Team
+ * All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package de.rwth.idsg.steve.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import de.rwth.idsg.steve.service.WebUserService;
+import de.rwth.idsg.steve.web.api.ApiControllerAdvice;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+/**
+ * @author Sevket Goekay <sevketgokay@gmail.com>
+ * @since 17.08.2024
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ApiAuthenticationManager implements AuthenticationManager, AuthenticationEntryPoint {
+
+    private final WebUserService webUserService;
+    private final PasswordEncoder passwordEncoder;
+    private final ObjectMapper jacksonObjectMapper;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String username = (String) authentication.getPrincipal();
+        String apiPassword = (String) authentication.getCredentials();
+
+        if (Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(apiPassword)) {
+            throw new BadCredentialsException("Required parameters missing");
+        }
+
+        UserDetails userDetails = webUserService.loadUserByUsernameForApi(username);
```

Comment:
this impl is a bit inefficient right now, since we go to database for each API call to validate the credentials

Reply:
added cache in https://github.com/steve-community/steve/pull/1545/commits/e646b474007cdfe9c0f5b931bf7718500a8f4373 with a TTL of 10 min